File: jvm_tests/tests/odd_even.rs
use jvm_tests::run_class;

// #[test]
pub fn test_odd_even() -> anyhow::Result<()> {
    let odd_even = include_bytes!("../../test_data/OddEven.class");

    let result = run_class("OddEven", odd_even, &["1234"])?;
    assert_eq!(result, "i is even\n");

    let result = run_class("OddEven", odd_even, &["1233"])?;
    assert_eq!(result, "i is odd\n");

    Ok(())
}

File: test_data/src/OddEven.java
class OddEven {
    public static void main(String[] args) {
        OddEven oe = new OddEven();

        System.out.println(oe.run("1234"));
        System.out.println(oe.run("1233"));
    }

    String run(String arg) {
        int i = Integer.parseInt(arg);

        if (i % 2 == 1) {
            return "i is odd";
        } else {
            return "i is even";
        }
    }
}
